### PR TITLE
updated docs

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,53 +156,50 @@ In the following i provide an example:
...
-            $galleryHasMedia->setGallery($gallery);
-            $galleryHasMedia->setMedia($media);
-            $gallery->addGalleryHasMedia($galleryHasMedia);
-        }
+public function createGalleryAction(Request $request): RedirectResponse
+{
+    $idx = $request->query->get('idx');
+    $idx = json_decode($idx);
+
+    $galleryManager = $this->get('sonata.media.manager.gallery');
```
Comment (OskarStark) [inline README.md]: this should be injected via DI

File: README.md [hunk 2]
```diff
@@ -154,53 +156,50 @@ In the following i provide an example:
...
+
+    $gallery = $galleryManager->create();
+    $gallery->setName('Auto Created Gallery');
+    $gallery->setEnabled(false);
+    $gallery->setContext('default');
+
+    $mediaManager = $this->get('sonata.media.manager.media');
```
Comment (OskarStark) [inline README.md]: this, too

Comment (silasjoisten) [reply README.md]: Good catch thank you!! @OskarStark

File: README.md [hunk 3]
```diff
@@ -154,53 +156,47 @@ In the following i provide an example:
...
+        $galleryHasMedia->setGallery($gallery);
+        $galleryHasMedia->setMedia($media);
+        $gallery->addGalleryHasMedia($galleryHasMedia);
+    }
 
-        $galleryManager->save($gallery);
+    $galleryManager->save($gallery);
 
-        return $this->redirect($this->get('sonata.media.admin.gallery')->generateObjectUrl('edit', $gallery));
-    }
```
Comment (OskarStark) [inline README.md]: You can use it here too

File: README.md [hunk 4]
```diff
@@ -154,53 +156,47 @@ In the following i provide an example:
...
-        $mediaManager = $this->get('sonata.media.manager.media');
-        foreach ($idx as $id) {
-            $media = $mediaManager->find($id);
-
-            $galleryHasMedia = new GalleryHasMedia();
-            $galleryHasMedia->setGallery($gallery);
-            $galleryHasMedia->setMedia($media);
-            $gallery->addGalleryHasMedia($galleryHasMedia);
-        }
+public function createGalleryAction(Request $request, MediaManager $mediaManager, GalleryManager $galleryManager): RedirectResponse
```
Comment (OskarStark) [inline README.md]: Did you test this is working? 

Maybe add use statements ?

Comment (silasjoisten) [reply README.md]: i mean for GalleryManager and MediaManager you need to create an alias like this before this works.

```yaml
  Sonata\MediaBundle\Entity\MediaManager:
        alias: sonata.media.manager.media

    Sonata\MediaBundle\Entity\GalleryManager:
        alias: sonata.media.manager.gallery

```

File: README.md [hunk 5]
```diff
@@ -151,56 +153,71 @@ a gallery from uploaded medias.
...
-     * @return RedirectResponse
-     */
-    public function createGalleryAction(Request $request)
+namespace App\Controller;
+
+use SilasJoisten\Sonata\MultiUploadBundle\Controller\MultiUploadController;
+use Sonata\MediaBundle\Entity\MediaManager;
+use Sonata\MediaBundle\Entity\GalleryManager;
+
+class MediaAdminController extends MultiUploadController
```
Comment (OskarStark) [inline README.md]: Great improvement of the example.

Consider making the controller class final.

Good to merge

File: README.md [hunk 6]
```diff
@@ -151,56 +153,75 @@ a gallery from uploaded medias.
...
     }
+}
+```
+
+Maybe you need to create an alias for `MediaManager` and `GalleryManager` like:
+```yaml
+# config/services.yaml
+services:
+    Sonata\MediaBundle\Entity\MediaManager:
+        alias: sonata.media.manager.media
```
Comment (OskarStark) [inline README.md]: Can you provide a PR to SonataMediaBundle for this?

File: README.md [hunk 7]
```diff
@@ -151,56 +153,75 @@ a gallery from uploaded medias.
...
+
+    Sonata\MediaBundle\Entity\GalleryManager:
+        alias: sonata.media.manager.gallery
```
Comment (OskarStark) [inline README.md]: Can you provide a PR to SonataMediaBundle for this?

Comment (OskarStark) [reply README.md]: An example is provided here

**Pull request merged**